src/Tools/ExternalAccess/Razor/RazorAnalyzerAssemblyResolver.cs

```diff
@@ -54,5 +55,12 @@ internal static bool TrySetAssemblyResolver(Func<AssemblyName, Assembly?> resolv
 
             return resolver?.Invoke(assemblyName);
         }
+
+        private class AssemblyNameEqualityComparer : IEqualityComparer<AssemblyName>
+        {
+            public bool Equals(AssemblyName? x, AssemblyName? y) => x?.FullName.Equals(y?.FullName) == true;
+
+            public int GetHashCode(AssemblyName obj) => Hash.GetFNVHashCode(obj.FullName);
```

Comment:
>obj [](http://example.com/codeflow?start=76&length=3)

Should we be prepared to deal with `null` as an input?

Reply:
`IEqualityComparer<T>.GetHashCode` is marked `[DisallowNull]` which I beleive the intent is that it will never be called with a null value. We also own the parent `IAnalyzerAssemblyResolver` interface, and this comparer is only used here, so annotations say we can never see a null value for `AssemblyName` even if the comparer did support calling it with a null value. I'm happy to handle null though if you'd prefer.